content/post/danielepo/pills/gestire-le-versioni-con-gitversion-introduzione.it.md

```diff
@@ -0,0 +1,102 @@
+Confrontare le "versioni" delle dll in base alla loro dimensione ed ipotizzare che quella più grossa sia la più precente perché "contiene più funzionalità"
+Solution così grosse che aprire tutti i progetti farebbe crashare il computer con tutti i problemi derivanti
+
+È chiaro che bisogna trovare un modo per gestire il versionamento. una serie di regole e convenzioni che aiutino gli sviluppatori a decidere se vale la pena passare alla versione successiva di una dipendenza o meno. 
+
+La convenzione più comunemente utilizzata al giorno d'oggi è detta semantic versioning o, per gli amici, SemVer e trovate tutte le informazioni approposito al sito semver.org, link nella descrizione del video. Io ora vi faccio un riassunto delle regole di base: 
+
+in pratica nella Semantic Versioning la versione è composta da tre numeri separati da punti detti rispettivamente Major, Minor e Patch, a cui è possibile accodare ettichette per aggiungere altre informazioni utili, per esempio versione uno punto cinque punto tre trattino beta
+
+la Semantic Versioning è una serie di regole su come incrementare i differenti valori Major, Minor e Patch in base a cosa è stato modifcato nel codice. In poche parole 
+
+Incrementate la Patch quando fate piccole modifiche e correzioni di bug che non influenzato come il mondo esterno interagisce con il vostro programma o libreria
+
+Incrementare la Minor quando fate modifiche o aggiunte funzionalità senza però rompere la compatibilità con le versioni precedenti
+
+ed infine, incrementate la Major quando fate modifiche che rendono parti del nuovo codice non più compatibile con le versioni precedenti e quindi che richiederebbero modifiche da parte dei programmi che dipendono dalla vostra libreria 
+
+Seguendo questa convenzione un utilizzatore della vostra libreria riesce a identificare a colpo d'occhio se può aggiornare la versione senza sorprese o se invece gli conviene leggere attentamente i Change Log.
+
+Per esempio facciamo finta che ho una libreria che fa i calcoli di conversione tra varie monete e la versione di questa libreria è "uno cinque tre", mi segnalano che una stringa ha un errore, quando lo correggo creerò la versione "uno cinque quattro", incrementando di uno la patch.
+
+se invece sto creando una nuova funzionalità per esempio aggiungo una moneta alla lista di monete tra cui posso fare la conversione dovrò incrementare la Minor azzerando allo stesso tempo la Patch, creo quindi la versione uno sei zero.
+
+adesso per qualche strano motivo invertiamo l'ordine in cui passo i dati alla funzione di calcolo, invece di essere quantità, moneta di partenza, moneta di cui voliamo sapere il valore mettiamo le due monete come primi parametri e la quantità la passiamo per ultimo. questa modifica romperebbe il codice dei nostri utilizzatori e richiederebbe loro di modificare il loro codice. per questa ragione aggiorneremmo la Major azzerando minor e patch allo stesso tempo. quindi la nostra versione diventa due zero zero.
```

Comment:
cambierei tutto in "ora, immaginiamo di dover invertire l'ordine dei parametri della funzione di calcolo. Tale modifica renderebbe non retrocompatibile il codice dei nostri utilizzatori e richiederebbe loro di modificare il codice. Per questa ragione aggiorneremmo la Major azzerando minor e patch allo stesso tempo. Quindi la nostra versione diventa due zero zero.